cmd/xl-v1-multipart.go

```diff
@@ -1204,5 +792,67 @@ func (xl xlObjects) AbortMultipartUpload(bucket, object, uploadID string) error
 	if !xl.isUploadIDExists(bucket, object, uploadID) {
 		return errors.Trace(InvalidUploadID{UploadID: uploadID})
 	}
-	return xl.abortMultipartUpload(bucket, object, uploadID)
+
+	// Read metadata associated with the object from all disks.
+	partsMetadata, errs := readAllXLMetadata(xl.getDisks(), minioMetaMultipartBucket, uploadIDPath)
+
+	// get Quorum for this object
+	_, writeQuorum, err := objectQuorumFromMeta(xl, partsMetadata, errs)
+	if err != nil {
+		return toObjectErr(err, bucket, object)
+	}
+
+	// Cleanup all uploaded parts.
+	if err = xl.cleanupUploadedParts(uploadIDPath, writeQuorum); err != nil {
+		return toObjectErr(err, bucket, object)
+	}
+
+	// Successfully purged.
+	return nil
+}
+
+// Clean-up the old multipart uploads. Should be run in a Go routine.
```

Comment:
we can't avoid taking in a channel, as it is the `globalServiceDoneCh` channel. Since the function will hang if the caller does not use `go` it will be evident when he runs minio as minio will just hang. Also this function should be called only from one place and doing `go xl.cleanupStaleMultipartUploads()` is an accepted pattern I think the current style is good.